cyral/resource_cyral_repository_access_rules.go

```diff
@@ -0,0 +1,335 @@
+					return &AccessRulesResponse{}
+				},
+			},
+			ReadRepositoryAccessRulesConfig,
+		),
+		ReadContext: ReadResource(ReadRepositoryAccessRulesConfig),
+		UpdateContext: UpdateResource(
+			ResourceOperationConfig{
+				Name:       "RepositoryAccessRulesUpdate",
+				HttpMethod: http.MethodPut,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+					return fmt.Sprintf("https://%s/v1/repos/%s/userAccounts/%s/accessRules",
+						c.ControlPlane,
+						d.Get("repository_id").(string),
+						d.Get("user_account_id").(string),
+					)
+				},
+				NewResourceData: func() ResourceData {
+					return &AccessRulesResource{}
+				},
+				NewResponseData: func(_ *schema.ResourceData) ResponseData {
+					return &AccessRulesResponse{}
+				},
+			},
+			ReadRepositoryAccessRulesConfig,
+		),
+		DeleteContext: DeleteResource(
+			ResourceOperationConfig{
+				Name:       "RepositoryAccessRulesDelete",
+				HttpMethod: http.MethodDelete,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+
+					idPieces, err := unmarshalComposedID(d.Id(), "/", 2)
+					if err != nil {
+						panic(fmt.Sprintf("Failed to unmarshal access rules ID: %v", err))
+					}
+					repoID := idPieces[0]
+					userAccountID := idPieces[1]
+
+					return fmt.Sprintf("https://%s/v1/repos/%s/userAccounts/%s/accessRules",
+						c.ControlPlane,
+						repoID,
+						userAccountID,
+					)
+				},
+			},
+		),
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description: "ID of this resource in Cyral environment.",
+				Computed:    true,
+				Type:        schema.TypeString,
+			},
+
+			"repository_id": {
+				Description: "ID of the repository.",
+				Required:    true,
+				Type:        schema.TypeString,
+				ForceNew:    true,
+			},
+
+			"user_account_id": {
+				Description: "ID of the database account.",
+				Required:    true,
+				Type:        schema.TypeString,
+				ForceNew:    true,
+			},
+
+			"rule": {
+				Description: "An ordered list of access rules.",
+				Required:    true,
+				Type:        schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+
+						"identity": {
+							Description: "The identity of the person/group getting access.",
+							Required:    true,
+							Type:        schema.TypeSet,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"type": {
+										Description: "Identity type. List of supported values: " +
+											supportedTypesMarkdown([]string{
+												"username",
+												"email",
+												"group",
+											}),
+										Required: true,
+										Type:     schema.TypeString,
+									},
+
+									"name": {
+										Description: "The name of the person/group getting access.",
+										Required:    true,
+										Type:        schema.TypeString,
+									},
+								},
+							},
+							MinItems: 1,
+							MaxItems: 1,
+						},
+
+						"valid_from": {
```

Comment:
We might want to make `valid_from` a required field depending on the conclusion of this [discussion](https://cyralinc.slack.com/archives/C0173L983EU/p1664240598935429).